services/middlewareServices/FormSGService.ts

```diff
@@ -0,0 +1,57 @@
+import FormSG from "@opengovsg/formsg-sdk"
+import { Request, Response, NextFunction } from "express"
+
+import logger from "@logger/logger"
+
+import { AuthError } from "@root/errors/AuthError"
+
+const formsg = FormSG()
+
+export default class FormSGService {
+  authenticate = () => (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): void => {
+    const signature = req.get("X-FormSG-Signature")
+    if (!signature) {
+      throw new AuthError("Signature missing")
+    }
+    const postUri = `https://${req.get("host")}${req.baseUrl}${req.path}`
```

Comment:
could i ask how this is derived? think i'm missing some context here

Reply:
This should be to generate the postURI forms needs to verify the submission - it was in Alwyn's existing code